models/workitemtype_repository.go

```diff
@@ -58,8 +58,13 @@ func (r *GormWorkItemTypeRepository) LoadTypeFromDB(ctx context.Context, name st
 // Create creates a new work item in the repository
 // returns BadParameterError, ConversionError or InternalError
 func (r *GormWorkItemTypeRepository) Create(ctx context.Context, extendedTypeName *string, name string, fields map[string]app.FieldDefinition) (*app.WorkItemType, error) {
+	existing, _ := r.LoadTypeFromDB(ctx, name)
+	if existing != nil {
+		log.Printf("creating type %s again", name)
+		return nil, BadParameterError{parameter: "name", value: name}
+	}
 	allFields := map[string]FieldDefinition{}
-	path := "/"
+	path := "/" + name
 	if extendedTypeName != nil {
 		extendedType := WorkItemType{}
 		if r.db.First(&extendedType, extendedTypeName).RecordNotFound() {
```

Comment:
Replace this

```go
if r.db.First(&extendedType, extendedTypeName).RecordNotFound() {
```

with 

```go
db := r.db.First(&extendedType, extendedTypeName)
if db.RecordNotFound() {
}
```

And then also check for `db.Error` to find non-`RecordNotFound` errors as well.